clouddriver-core/src/test/java/com/netflix/spinnaker/clouddriver/safety/TrafficGuardTest.java

```diff
@@ -0,0 +1,134 @@
+import com.netflix.spinnaker.kork.retrofit.Retrofit2ServiceFactory;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerHttpException;
+import com.netflix.spinnaker.moniker.Moniker;
+import com.netflix.spinnaker.okhttp.OkHttpClientConfigurationProperties;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestHeaderInterceptor;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import okhttp3.OkHttpClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+
+@SpringBootTest(
+    classes = {
+      RetrofitConfig.class,
+      OkHttpClientProvider.class,
+      TrafficGuardTest.TestConfig.class,
+      DefaultServiceClientProvider.class,
+      Retrofit2ServiceFactory.class,
+      DefaultOkHttpClientBuilderProvider.class,
+      OkHttpClient.class,
+      OkHttpClientConfigurationProperties.class,
+      ObjectMapper.class
+    })
+public class TrafficGuardTest {
+  private final String APP_NAME = "testApp";
+
+  @RegisterExtension
+  static WireMockExtension wmFront50 =
+      WireMockExtension.newInstance().options(wireMockConfig().dynamicPort()).build();
+
+  @Autowired ObjectMapper objectMapper;
+
+  @Autowired Front50Service front50Service;
+
+  @Autowired SpinnakerRequestHeaderInterceptor requestHeaderInterceptor;
+
+  TrafficGuard trafficGuard;
+
+  @DynamicPropertySource
+  static void registerUrls(DynamicPropertyRegistry registry) {
+    // Configure wiremock's random ports into clouddriver
+    System.out.println("wiremock front50 url: " + wmFront50.baseUrl());
+    registry.add("services.front50.base-url", wmFront50::baseUrl);
+  }
+
+  @BeforeEach
+  void init() throws JsonProcessingException {
+    wmFront50.stubFor(
+        WireMock.get(urlMatching("/v2/applications/" + APP_NAME))
+            .willReturn(
+                aResponse()
+                    .withStatus(HttpStatus.INTERNAL_SERVER_ERROR.value())
+                    .withBody(
+                        objectMapper.writeValueAsString(
+                            Map.of("error", "Internal Server Error")))));
+
+    trafficGuard =
+        new TrafficGuard(
+            List.of(new NoopClusterProvider()),
+            Optional.of(front50Service),
+            new NoopRegistry(),
+            new DynamicConfigService.NoopDynamicConfig());
+  }
+
+  @Test
+  void verifyErrorHandlingCallAdapterFactory_with_500_exception() {
+    Moniker moniker = Moniker.builder().app(APP_NAME).build();
+    Exception ex = null;
```

Comment:
Please upgrade this to a fancier assert mechanism, so we get a nicer message when it fails.  Perhaps (untested):
```
import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;

assertThatThrownBy(() -> trafficGuard.hasDisableLock(moniker, "acc1", "location1")).isInstanceOf(SpinnakerHttpException.class);
```

Reply:
done